lib/components/ScheduleSelector/components/IntervalSelector/IntervalSelector.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, FC, useCallback } from 'react'
+import { TimePicker, DayPicker } from '../index'
+
+import './intervalSelector.scss'
+import '../../scheduleSelector.scss'
+
+interface IntervalSelectorProps {
+  intervalData: {
+    days?: string
+    start_time?: string
+    end_time?: string
+    interval?: number
+  }
+  onChange: (data: Record<string, any>) => void
+  isDisabled?: boolean
+}
+
+const IntervalSelector: FC<IntervalSelectorProps> = ({
+  intervalData,
+  onChange,
+  isDisabled
+}) => {
+  const [startTime, setStartTime] = useState(intervalData.start_time || '')
+  const [endTime, setEndTime] = useState(intervalData.end_time || '')
+
+  const days =
+    intervalData.days || 'monday, tuesday, wednesday, thursday, friday'
+
+  const handleStartTimeChange = useCallback(
+    (time: string) => {
+      setStartTime(time)
+      const interval = intervalData.interval || 30
+      const [hours, minutes] = time.split(':').map(Number)
+      const newMinutes = minutes + interval
+
+      const adjustedHours = (hours + Math.floor(newMinutes / 60)) % 24
+      const adjustedMinutes = newMinutes % 60
+
+      setEndTime(
+        `${String(adjustedHours).padStart(2, '0')}:${String(
+          adjustedMinutes
+        ).padStart(2, '0')}`
+      )
+      onChange({
+        days,
+        interval,
+        start_time: time,
+        end_time: endTime
+      })
+    },
+    [intervalData.interval, days, endTime, onChange]
+  )
+
+  const handleChangeInterval = useCallback(
+    (interval: number) => {
+      handleStartTimeChange(startTime)
+      onChange({
+        days,
+        interval,
+        start_time: startTime,
+        end_time: endTime
+      })
+    },
+    [days, endTime, handleStartTimeChange, onChange, startTime]
+  )
+
+  const handleChangeDays = useCallback(
+    (days: string) => {
+      onChange({
+        days,
+        interval: intervalData.interval || 30,
+        start_time: startTime,
+        end_time: endTime
+      })
+    },
+    [intervalData.interval, startTime, endTime, onChange]
+  )
+
+  const handleEndTimeChange = useCallback(
+    (time: string) => {
+      setEndTime(time)
+      onChange({
+        days,
+        interval: intervalData.interval || 30,
+        start_time: startTime,
+        end_time: time
+      })
+    },
+    [days, intervalData.interval, startTime, onChange]
+  )
+
+  return (
+    <div className='intervalSelector-container'>
+      <span className='label-2'>Every</span>
+      <div className='interval-value'>
+        <input
+          type='number'
+          disabled={isDisabled}
+          value={intervalData.interval || 30}
+          onChange={(e) => handleChangeInterval(Number(e.target.value))}
```

Comment:
I guess we should limit it here to be between 0 and 60(59?)? or am I missing smth?

Reply:
Yes, there can be more validation rules, but for now I'll limit it to 59